### PR TITLE
Add bottom bar to the home menu

### PR DESCRIPTION
**Pull request by Vanethos**
Add bottom bar to the home menu with:
- Callbacks for each button
- Each button click is wired to the BLoC
- A pop up menu

Added colors file
Added dimmensions file
Added text styles file
Added primary and accent color to app

#closes 24

### CODE REVIEW

File: lib/presentation/assets/colors.dart [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import 'dart:ui';
+
+const colorPrimary = Color(0xFF7CB342);
+const colorAccent = Color(0xFFFBC02D);
```
Comment (miguelpruivo) [inline lib/presentation/assets/colors.dart:4]: **Sugestão:** Não será boa ideia extrairmos isto para uma classe que possa fazer a gestão do tema? Mesmo que sejam cores estáticas.

**Review by miguelpruivo — APPROVED** [review]

**Review by miguelpruivo — APPROVED** [review]

**Review by miguelpruivo — APPROVED** [review]

File: lib/presentation/assets/colors.dart (hunk 1 above)
Comment (Vanethos) [reply lib/presentation/assets/colors.dart:4]: I think that could be a nice solution, indeed, should we open a new issue for that?

File: lib/presentation/ui/home/home_page.dart [hunk 2]
```diff
@@ -20,32 +26,120 @@ class _MyHomePageState extends BaseState<HomePage> {
...
+                          children: <Widget>[
+                            IconButton(
+                              padding: EdgeInsets.only(
+                                  left: marginSmall,
+                                  right: marginSmall,
+                                  top: marginSmall),
+                              icon: Icon(snapshot.data == HomeBloc.recentsIndex
+                                  ? Icons.list
+                                  : Icons.star),
+                              // since the onClick is handled by another widget, this
```
Comment (miguelpruivo) [inline lib/presentation/ui/home/home_page.dart:92]: Aqui sugiro colocar o `onPressed: null` para evitar warnings (apesar de ser redundante).

File: lib/presentation/utils/misc.dart [hunk 3]
```diff
@@ -6,4 +6,11 @@ void showErrorSnackbar(String event, ScaffoldState context) {
       content: new Text(event),
     ));
   }
-}
\ No newline at end of file
+}
+
+double findTextWidth(String text, TextStyle style) {
+  final textPainter = TextPainter(textDirection: TextDirection.ltr);
+  textPainter.text = TextSpan(text: text, style: style);
```
Comment (miguelpruivo) [inline lib/presentation/utils/misc.dart:13]: Mudar aqui o `text` para `text.toUpperCase()`.

File: lib/presentation/ui/home/home_page.dart [hunk 4]
```diff
@@ -20,32 +26,120 @@ class _MyHomePageState extends BaseState<HomePage> {
...
+                initialData: widget.bloc.currentTypeOfDataSubject.value,
+                stream: widget.bloc.currentTypeOfDataStream,
+                builder: (context, snapshot) {
+                  return Container(
+                    margin: EdgeInsets.only(right: marginSmall),
+                    child: InkWell(
+                      onTap: _onToggleTypeTap,
+                      child: SizedBox(
+                        // we find the biggest possible string and give it a margin
+                        width: _findBiggestTextWidth() * 1.5,
```
Comment (miguelpruivo) [inline lib/presentation/ui/home/home_page.dart]: Remover o `*1.5`.

**Review by miguelpruivo — CHANGES_REQUESTED** [review]
Ver comentários.

File: lib/presentation/assets/colors.dart (hunk 1 above)
Comment (miguelpruivo) [reply lib/presentation/assets/colors.dart:4]: Yes, that could fit in a low priority issue.

**Comment by Vanethos** [comment]
Fixed comments

**Review by miguelpruivo — APPROVED** [review]

**Pull request merged**